examples/advance/main.tf

```diff
@@ -0,0 +1,52 @@
+##############################################################################
+# Resource Group
+##############################################################################
+
+module "resource_group" {
+  source  = "terraform-ibm-modules/resource-group/ibm"
+  version = "1.1.4"
+  # if an existing resource group is not set (null) create a new one using prefix
+  resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
+  existing_resource_group_name = var.resource_group
+}
+
+##############################################################################
+# Key Protect with Private Service Endpoint and Keyring
+##############################################################################
+
+module "key_protect_module" {
+  source            = "../.."
+  key_protect_name  = "${var.prefix}-kp"
+  resource_group_id = module.resource_group.resource_group_id
+  region            = var.region
+  tags              = var.resource_tags
+  access_tags       = var.access_tags
+  #service_endpoints  = "private-only"      #Uncomment this line to create the Private Key Protect, this will require you to be part of IBM cloud private network when creating this resource
```

Comment:
set this to `service_endpoints = "private"` - we can write a test that actually tests this test in IBM Schematics which will have access to the private network (see subsequent comment)